docs/config.rst

```diff
@@ -9,33 +9,47 @@ be set once and used for all tox environments, while environment options are app
 Discovery and file types
 ------------------------
 
-Out of box tox supports three configuration locations prioritized in the following order:
+With regards to the configuration format, at the moment we support the following formats:
 
-1. ``tox.ini``,
-2. ``pyproject.toml``,
-3. ``setup.cfg``.
+- `INI <https://en.wikipedia.org/wiki/INI_file>`_.
+- `TOML <https://toml.io/en/v1.0.0>`_.
 
-With regards to the configuration format, at the moment we only support *ini-style*. ``tox.ini`` and ``setup.cfg`` are
-by nature such files, while in ``pyproject.toml`` currently you can only inline the *ini-style* config.
+Out of box tox supports five configuration locations prioritized in the following order:
 
-Note that ``setup.cfg`` requires the content to be under the ``tox:tox`` and ``testenv`` sections and is otherwise
-ignored. ``pyproject.toml`` on the other hand is in TOML format. However, one can inline the *ini-style* format under
-the ``tool.tox.legacy_tox_ini`` key as a multi-line string.
+1. ``tox.ini`` (INI),
+2. ``setup.cfg`` (INI),
+3. ``pyproject.toml`` with the ``tool.tox`` table, having ``legacy_tox_ini`` key (containing INI),
+4. Native ``pyproject.toml`` under the ``tool.tox`` table (TOML),
+5. ``tox.toml`` (TOML).
+
+Historically, the INI format was created first, and TOML was added in 2024. The TOML format generally is more robust,
+however is less powerful and more verbose. You should use TOML unless you need some of the more advanced features
+that TOML does not support (such as conditional factors, generative environments to name a few -- however, PRs to
+add support for these in TOML is welcome).
```

Comment:
I'd mark TOML config as experimental until all these are supported. Supporting those may affect the TOML file structure.

Reply:
I disagree. Some of these features I explicitly do not want to support in the new format (such as conditional factors). The TOML file Format is just not a good match for it.